activerecord/lib/active_record/connection_adapters/abstract/lazy_transaction_proxy.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module ActiveRecord
+  module ConnectionAdapters
+    class LazyTransactionProxy < SimpleDelegator # :nodoc:
+      class StatementProxy < SimpleDelegator
```

Comment:
Does it make sense to use `DelegateClass` instead for both of these?

Reply:
I don't think so, because the type of object they wrap depends on the adapter in use; there's no single class we can provide with the full set of methods they need to support.

Reply:
This comment did make me realise that I forgot to `:nodoc:` the inner class, though.